internal/driver/db.go

```diff
@@ -1,142 +1,117 @@
 package driver
 
 import (
-	"database/sql"
-	"fmt"
-	"strconv"
-)
+	"sync"
 
-const (
-	qlDatabaseDriverName = "ql2"
-	qlDatabaseName       = "memory://deviceVirtual.db"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
 )
 
-var data struct {
-	DeviceName          string
+type data struct {
 	CommandName         string
-	DeviceResourceName  string
-	EnableRandomization string
+	EnableRandomization bool
 	DataType            string
 	Value               string
 }
 
+// Global: the data we are tasked with storing
+// Outer key: device name, inner key: resource name, value: resource info
+var resources map[string]map[string]data
+
+// Golang maps alone are not safe for concurrent access
+var resources_lock sync.RWMutex
```

Comment:
The `resources` and `resources_lock` should be moved to the db struct
avoid to use the global variables

Reply:
Brought them into the struct and it still works.